src/main/java/com/microsoft/cse/helium/app/services/keyvault/EnvironmentReader.java

```diff
@@ -0,0 +1,179 @@
+      this.authType = Constants.USE_MSI;
+    } else if (authType.equals(Constants.USE_CLI)) {
+      this.authType = Constants.USE_CLI;
+    } else if (authType.equals(Constants.USE_VS)) {
+      System.out.println("VS Credentials are not yet supported in Java");
+      System.exit(-1);
+    } else {
+      System.out.println("Usage: mvn clean spring-boot:run  "
+          + "-Dspring-boot.run.arguments=\"--h --auth-type=<CLI|MSI|VS>"
+          + " -keyvault-name=<keyVaultName>\"");
+      System.exit(-1);
+    }
+  }
+
+  /**
+   * getAuthType.
+   *
+   * @return returns CLI or MSI.
+   */
+  @SuppressFBWarnings("DM_EXIT")
+  public String getAuthType() {
+    if (this.authType != null) {
+      return this.authType;
+    }
+
+    authType = System.getenv(Constants.AUTH_TYPE);
+
+    // If it is not set, use the MSI
+    if (authType == null) {
+      return Constants.USE_MSI;
+    }
+
+    // ONLY If it is set and values are either MSI or CLI, we will accept.
+    // otherwise, default is just the MSI
+    if (authType.equals(Constants.USE_MSI)) {
+      return Constants.USE_MSI;
+    } else if (authType.equals(Constants.USE_CLI)) {
+      return Constants.USE_CLI;
+    } else if (authType.equals(Constants.USE_VS)) {
+      System.out.println("VS Credentials are not yet supported in Java");
+      System.exit(-1);
+      // following return needs to be there because java compiler wants a return statement.
+      return null;
+    } else {
+      System.exit(-1);
+      // following return needs to be there because java compiler wants a return statement.
+      return null;
+    }
+  }
+
+  @SuppressFBWarnings("DM_EXIT")
+  private void setKeyVaultName(String kvName) {
+    if (kvName == null) {
+      System.out.println("Usage: mvn clean spring-boot:run  "
+          + "-Dspring-boot.run.arguments=\"--h --auth-type=<CLI|MSI|VS>"
+          + " -keyvault-name=<keyVaultName>\"");
+      System.exit(-1);
+    }
+
+    if (!isValidKeyVaultName(kvName)) {
+      logger.error("helium.keyvault.name (KeyVaultName) value is '" + kvName
+          + "' which does not meet the criteria must be 3-24 characters long, begin with a "
+          + "character, may contain alphanumeric or hyphen, no repeating hyphens, and end with "
+          + "alphanumeric.  Check ${KeyVaultName} in your environment variables.");
+      System.out.println("Usage: mvn clean spring-boot:run  "
+          + "-Dspring-boot.run.arguments=\"--h --auth-type=<CLI|MSI|VS>"
+          + " -keyvault-name=<keyVaultName>\"");
+      System.exit(-1);
+    }
+
+    this.keyVaultName = kvName;
+  }
+
+  /**
+   * getKeyVaultName.
+   *
+   * @return returns the key vault name.
+   */
+  @SuppressFBWarnings("DM_EXIT")
+  public String getKeyVaultName() {
+    if (this.keyVaultName != null) {
+      return this.keyVaultName;
+    }
+
+    keyVaultName = System.getenv(Constants.KEY_VAULT_NAME);
+    if (keyVaultName == null || !isValidKeyVaultName(keyVaultName)) {
+      System.exit(-1);
```

Comment:
does this need a return null too?

Reply:
This is an issue with the java compiler. It does not  understand that a System.exit is a termination of the program and expects a return statement - does not compile without it.

Reply:
I understand that, but it doesn't have a return null like the other instances do.

Also note that calling exit outside of main makes unit testing very challenging. Might make sense to bubble a result up to main and only exit from there.